eo/src/eoRanking.h

```diff
@@ -40,73 +40,85 @@ template <class EOT>
 class eoRanking : public eoPerf2Worth<EOT> // false: do not cache fitness
 {
 public:
-
     using eoPerf2Worth<EOT>::value;
 
-  /* Ctor:
-   @param _p selective pressure (in (1,2]
-   @param _e exponent (1 == linear)
-  */
-  eoRanking(double _p=2.0, double _e=1.0):
-    pressure(_p), exponent(_e) {}
+    /* Ctor:
+     @param _p selective pressure (in (1,2]
+     @param _e exponent (1 == linear)
+    */
+    eoRanking(double _p = 2.0, double _e = 1.0) : pressure(_p), exponent(_e)
+    {
+        if (pressure <= 1.0)
+        {
+            std::string msg = "eoRanking: pressure must be > 1.0";
+            eo::log << eo::errors << "ERROR: " << msg << std::endl;
+            throw eoException(msg);
+        }
```

Comment:
I understand the idea of having a consistent errors managements. However, since we are paying attention to performances, we don't want to add too many tests for states that will never occur in experiment conditions. Thus, parameters that should be all set up should be tested with `assert`, which is declutched in `Debug` builds, saving (overall *a lot* of) tests.

Exceptions are to be used for checking states that we cannot manage easily from high-level interfaces, for instance variables that changes during computations.